pykern/pkcollections.py

```diff
@@ -211,35 +211,62 @@ def pknested_set(self, qualifiers, value):
         return self
 
     def pksetdefault(self, *args, **kwargs):
-        """Get value or set it, possibly after evaluating arg.
+        """Set default values for keys in pairs
 
         Must pass an even number of args or kwargs, but not both. Each pair
         is interpreted as (key, value).
 
-        If self does not have `key`, then it will be set. If `value` is a callable,
-        it will be called to get the value to set.
+        If self does not have `key`, then it will be set to
+        `value`. If `value` is a callable, it will be called, and
+        `key` will be set to the returned value.
 
-        Values will be called if they are callable
+        A single value will be called if they are callable
 
         Args:
             key (object): value to get or set
             value (object): if callable, will be called, else verbatim
         Returns:
             object: self
         """
-        assert not (args and kwargs), "one of args or kwargs must be set, but not both"
```

Comment:
Why do you prefer `raise AssertionError` over `assert`?

Reply:
This is not well-documented. It should be in [DesignHints](https://github.com/radiasoft/pykern/wiki/DesignHints). Here's the discussion https://github.com/radiasoft/sirepo/issues/3592